### PR TITLE
Size class

### PR DESCRIPTION
**Pull request by bcarnevale-myob**

### CODE REVIEW

File: src/Field/Field.java [hunk 1]
```diff
@@ -3,40 +3,39 @@
...
 
-    public Field(int row, int col) {
-        this.field = new Square[row][col];
-        for(int x = 0; x < row; x++) {
-            for(int y = 0; y < col; y++) {
+    public Field(Size size, MinePlacer minePlacer) {
+        int height = size.getHeight();
+        int width = size.getWidth();
+        this.minePlacer = minePlacer;
+        this.field = new Square[height][width];
```
Comment (sol-myob) [inline src/Field/Field.java:12]: This is ok, but you could just put size.getHeight(), size.getWidth() in the square brackets

File: src/Field/Field.java [hunk 2]
```diff
@@ -3,40 +3,39 @@
...
+
+        for(int x = 0; x < height; x++) {
+            for(int y = 0; y < width; y++) {
```
Comment (sol-myob) [inline src/Field/Field.java:15]: you can use size.Get* here too

File: test/FieldTests/RandomiseMines.java [hunk 3]
```diff
@@ -2,13 +2,14 @@
...
 import org.junit.jupiter.api.Test;
+
 import static org.junit.jupiter.api.Assertions.*;
 
 public class RandomiseMines {
 
     @Test
     public void ifAFieldIsCreatedWithRandomMinePlacementTheFieldWillNotBeEmpty() {
-        Field field = new Field(5,5, new MockMineRandom(1, 0, 0));
+        Field field = new Field(new Size(5,5), new RandomMinePlacer(5, 5));
```
Comment (sol-myob) [inline test/FieldTests/RandomiseMines.java:12]: It's possible that the constructor of RandomMinePlace can also take in a size

Comment (nskoufis) [inline test/FieldTests/RandomiseMines.java:12]: Remove this test and write proper tests for random mine placer, but in a new branch.

**Review by nskoufis — APPROVED** [review]
LGTM, stuff with random mine placement will go in a new branch

**Pull request merged**